Google.PowerShell.IntegrationTests/Storage/Storage.GcsObject.Tests.ps1

```diff
@@ -207,6 +212,66 @@ Describe "New-GcsObject" {
         $obj.Metadata["Content-Type"] | Should Be "image/jpeg"
         Remove-GcsObject $obj
     }
+
+    It "should upload an empty folder" {
+        $TestFolder = Join-Path $TestDrive "TestFolder"
+
+        try {
+            if (-not (Test-Path $TestFolder))
+            {
+                New-Item -ItemType Directory -Path $TestFolder | Out-Null
+            }
+
+            $folder = New-GcsObject -Bucket $bucket -Folder $TestFolder -Force
+            $folderName = "TestFolder/"
+
+            $folder.Name | should be $folderName
+            $folder.Size | should be 0
+
+            $folderOnline = Get-GcsObject -Bucket $bucket -ObjectName $folderName
+            $folderOnline.Name | should be $folderName
+            $folderOnline.Size | should be 0
+        }
+        finally
+        {
+            Remove-Item $TestFolder -Recurse -Force -ErrorAction Ignore
+        }
+    }
+
+    It "should upload a folder with files and subfolders" {
+        $TestFolder = Join-Path $TestDrive "TestFolder"
+
+        try {
+            if (-not (Test-Path $TestFolder))
+            {
+                New-Item -ItemType Directory -Path $TestFolder | Out-Null
+            }
+
+            "Hello, world" | Out-File (Join-Path $TestFolder "world.txt")
```

Comment:
IMHO, it would be cleaner to just write `"$testFolder/world.txt"`. I understand `Join-Path` makes sense, but in this case we can be more fast and loose since it would aid readability.